pages/posts/github-case-study.md

```diff
@@ -0,0 +1,80 @@
+
+Without proper schemas for their JSON data, there was no way to verify whether code changes introduced new bugs, or to ensure that data consumed from external sources was in the format needed for automated documentation pages.
+
+<figure className="mt-10">
+  <img src="/blog/img/posts/2023/github-case-study/working.webp"/>
+  <figcaption className="mt-2 mb-10 text-sm text-center text-gray-500">Hubbers (GitHub staff) working to empower developers.</figcaption>
+</figure>
+
+## Solution
+
+The team began to introduce JSON Schemas to validate all of the JSON data files, context data and API request bodies consumed or produced as part of the application.
+Each time a change to the data model is made, the schemas are updated accordingly, and in any circumstance the schemas are used to validate that the application functions correctly with each change.
+
+This occurs in three main ways:
+
+  * While the application runs in production, each API call has its request body validated via a corresponding JSON Schema before passing the event off to a data warehouse
+  * When retrieving external data in automation pipelines, whenever data is transformed from a source format into JSON, the generated data is validated against a JSON Schema to ensure it has been correctly transformed.
+    If validation succeeds, the generated data is checked into a git repository to be used in production.
+    Otherwise, the failure is raised for investigation.
+  * When running continuous integration each time a change is made to the application, various additional schemas ensure:
+
+    * that YAML frontmatter properties are correctly included in Markdown files which are used to generate pages within the application
+    * that YAML or JSON data files which contain page content and which have been hand authored by content writers are correctly formed
+    * that the context object created at runtime which contains the entirety of the site content along with a site tree for the application is itself correctly formed
+
+<p className="text-2xl my-10">"Choosing JSON schema to allow JSON schema validation was a natural and obvious choice that our team made. It has been a fundamental part of our application since we moved from having a static site to a dynamic application about 3 years ago." - Rachael Sewell & Robert Sese, Docs Engineers at GitHub</p>
+
+## Impact
+
+Introducing JSON Schema into the platform produced meaningful impact in productivity, in discoverability, as well as in reliability.
+
+<p className="text-2xl my-10">"JSON schema makes it so much easier to see the shape of a data and its property types. I can quickly open the file on disk and understand what the data structure looks like. This saves the whole team time when extending a feature that relies on data backed by a schema." - Rachael Sewell & Robert Sese, Docs Engineers at GitHub</p>
```

Comment:
Is it necessary to restate Rachael's qualifications (see line 54).  Typically I see secondary quotes attributed using just the last name.  (We need a professional editor 😄.)

Reply:
I am of course no professional editor but my suspicion is (even besides style being a matter mostly of readability) that it might also differ given the first quote was a blockquote (as opposed to being truly inline in the text) which are more like "highlights" that in theory someone may have only glanced over. But clearly no idea, I'd probably leave it though unless you really think it matters.

Reply:
It just feels redundant to me to have it in both places. Leave it if you think it's fine.

Reply:
I chocse this pattern because it makes the block quotes more copyable. I've seen this done many times in other press release style articles.

Block quotes may be read before the rest of the article as someone scans it to see if it's worthwhile. IMHO, it shouldn't be considered really part of the article in terms of having all the context of the prior content.